src/App.js

```diff
@@ -1,21 +1,32 @@
-import {useState} from "react"
+import {useEffect, useState} from "react"
 import React from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem("savedTodoList")));
+  
+  useEffect(() => {
+    localStorage.setItem("savedTodoList", JSON.stringify(todoList))}, [todoList]
+  );
+
+  return [todoList, setTodoList];
+}
+
 function App() {
-  const [todoList, setTodoList] = React.useState([]);
+  const [todoList, setTodoList] = useSemiPersistentState();
   
   function addTodo(newTodo) {
     setTodoList([...todoList, newTodo])
```

Comment:
So this is my fault -- After more research, I found the way we set the new array was wrong. React allows us to provide a function to the setter (setTodoList) and function param will get the most recent value of the state.

So it'd look something like this:
```suggestion
  const addTodo = (newTodo) => {
    setTodoList([...todoList, newTodo]);
  };
  const addTodo = (newTodo) => {
     setTodoList( (previousTodoList) => [...previousTodoList, newTodo])
  };
```
I used an arrow function there b/c it's easier to read. How does all this sound to you?

Exercise 1.4 overview section actually explains my comment a bit better 😆
https://github.com/Code-the-Dream-School/react-cockatoo/wiki/Lesson-1.4
take a look